tools/booster/Proxy.hs

```diff
@@ -219,6 +219,61 @@ respondEither mbStatsVar booster kore req = case req of
             -- can only be an error at this point
             res -> pure res
 
+    simplifyExecuteState :: Maybe Text -> ExecuteState -> m ExecuteState
+    simplifyExecuteState mbModule s = do
+        Log.logInfoNS "proxy" "Simplifying execution state"
+        simplResult <-
+            handleSimplify $ toSimplifyRequest s
+        case simplResult of
+            -- This request should not fail, as the only possible
+            -- failure mode would be malformed or invalid kore
+            Right (Simplify simplified) -> do
+                -- to convert back to a term/constraints form,
+                -- we run a trivial execute request (in booster)
+                -- We cannot call the booster internaliser without access to the server state
+                -- Again this should not fail.
+                let request =
+                        (emptyExecuteRequest simplified.state)
+                            { _module = mbModule
+                            , maxDepth = Just $ Depth 0
+                            }
+                Log.logInfoNS "proxy" "Making 0-step execute request to convert back to a term/constraints form"
+                result <- booster $ Execute request
+                case result of
+                    Right (Execute ExecuteResult{state = finalState}) -> pure finalState
+                    _other -> pure s
+            _other ->
+                -- TODO log error
+                pure s -- if we hit an error here, return the original
+      where
+        toSimplifyRequest :: ExecuteState -> SimplifyRequest
+        toSimplifyRequest ExecuteState{term = t, substitution, predicate} =
+            let subAndPred = catMaybes [KoreJson.term <$> substitution, KoreJson.term <$> predicate]
+                termSort = KoreJson.SortApp (KoreJson.Id "SortGeneratedTopCell") []
+                term = t{KoreJson.term = foldr (KoreJson.KJAnd termSort) t.term subAndPred}
```

Comment:
This is basically `Booster.JsonRpc.execStateToKore`.
What occurred to me while reading that function is that we should try to use the sort of `t` instead of `SortGeneratedTopCell`.  We have `Booster.Syntax.Json.sortOfJson` but that one cannot give a result for `app` nodes. In principle we should check that all sorts in the `subAndPred` list are the same and use that, unless empty.